immuneML/reports/ml_reports/Evaluations.py

```diff
@@ -0,0 +1,217 @@
+from pathlib import Path
+import warnings
+import yaml
+
+from immuneML.data_model.dataset.Dataset import Dataset
+from immuneML.environment.Metric import Metric
+from immuneML.hyperparameter_optimization import HPSetting
+from immuneML.ml_methods.MLMethod import MLMethod
+from immuneML.ml_metrics import ml_metrics
+from immuneML.reports.ReportOutput import ReportOutput
+from immuneML.reports.ReportResult import ReportResult
+from immuneML.reports.ml_reports.MLReport import MLReport
+from immuneML.util.ParameterValidator import ParameterValidator
+from immuneML.util.PathBuilder import PathBuilder
+from sklearn import metrics as sklearn_metrics
+from sklearn.preprocessing import label_binarize
+
+import numpy as np
+import pandas as pd
+import plotly.graph_objs as go
+
+class Evaluations(MLReport):
+    """
+    A report that plots the evaluation metrics for the performance given machine learning model and testing dataset.
+    The available metrics are accuracy, balanced_accuracy, confusion_matrix, f1_micro, f1_macro, f1_weighted, precision,
+    recall, auc and log_loss (see :ref:`Metrics`).
+
+    Arguments:
+
+        metrics (list): A list of evaluation metrics to compute. See :ref:`Metrics` for available options.
+
+    YAML specification:
+
+    .. indent with spaces
+    .. code-block:: yaml
+
+        my_eval_report:
+            Evaluations: 
+                metrics:
+                    - accuracy
+                    - balanced_accuracy
+                    - confusion_matrix
+                    - f1_micro
+                    - f1_macro
+                    - f1_weighted
+                    - precision
+                    - recall
+                    - auc
+                    - log_loss
+
+    """
+
+    @classmethod
+    def build_object(cls, **kwargs):
+        location = "Evaluations"        
+        valid_metrics = [m.name for m in Metric]
+
+        name = kwargs["name"] if "name" in kwargs else None
+        metrics = kwargs["metrics"] if "metrics" in kwargs else valid_metrics
+        metrics = [m.upper() for m in metrics]
+        
+        ParameterValidator.assert_all_in_valid_list(metrics, valid_metrics, location, 'metrics')
+
+        return Evaluations(metrics, name=name)
+
+    def __init__(self, metrics_set: set, train_dataset: Dataset = None, test_dataset: Dataset = None, method: MLMethod = None,
+                 result_path: Path = None, name: str = None, hp_setting: HPSetting = None):
+        super().__init__(train_dataset, test_dataset, method, result_path, name, hp_setting)
+        self.metrics_set = set(metrics_set)
+
+    def _generate(self) -> ReportResult:
+        
+        X = self.test_dataset.encoded_data
```

Comment:
minor edit: this should all be computed on the training data (for the test data it is already computed) :)